tests/test_modeling.py

```diff
@@ -0,0 +1,913 @@
+
+        self.assertIsInstance(model.model, torch.jit._script.ScriptModule)
+        self.assertIsInstance(model.config, PretrainedConfig)
+
+    def test_load_model_from_empty_cache(self):
+        dirpath = os.path.join(default_cache_path, "models--" + self.NEURON_MODEL_ID.replace("/", "--"))
+
+        if os.path.exists(dirpath) and os.path.isdir(dirpath):
+            shutil.rmtree(dirpath)
+        with self.assertRaises(Exception):
+            _ = NeuronModel.from_pretrained(self.NEURON_MODEL_ID, local_files_only=True)
+
+    def test_load_model_from_hub_without_neuron_model(self):
+        with self.assertRaises(FileNotFoundError):
+            NeuronModel.from_pretrained(self.FAIL_NEURON_MODEL_ID)
+
+    @require_hf_token
+    def test_load_model_from_hub_private(self):
+        model = NeuronModel.from_pretrained(
+            self.PRIVATE_NEURON_MODEL_ID, use_auth_token=os.environ.get("HF_AUTH_TOKEN", None)
+        )
+        self.assertIsInstance(model.model, torch.jit._script.ScriptModule)
+        self.assertIsInstance(model.config, PretrainedConfig)
+
+    def test_save_model(self):
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            model = NeuronModel.from_pretrained(self.LOCAL_MODEL_PATH)
+            model.save_pretrained(tmpdirname)
+            # folder contains all config files and neuron exported model
+            folder_contents = os.listdir(tmpdirname)
+            self.assertTrue(NEURON_FILE_NAME in folder_contents)
+            self.assertTrue(CONFIG_NAME in folder_contents)
+
+    @require_hf_token
+    def test_push_model_to_hub(self):
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            model = NeuronModel.from_pretrained(self.LOCAL_MODEL_PATH)
+            model.save_pretrained(
+                tmpdirname,
+                use_auth_token=os.environ.get("HF_AUTH_TOKEN", None),
+                push_to_hub=True,
+                repository_id=self.PRIVATE_NEURON_MODEL_ID,
+                private=True,
+            )
+
+    def test_trust_remote_code(self):
+        model = NeuronModelForSequenceClassification.from_pretrained(
+            self.TINY_MODEL_REMOTE, export=True, trust_remote_code=True, **self.INPUTS_SHAPES
+        )
+        self.assertIsInstance(model.model, torch.jit._script.ScriptModule)
+        self.assertIsInstance(model.config, PretrainedConfig)
+
+
+@is_inferentia_test
+class NeuronModelForFeatureExtractionIntegrationTest(NeuronModelTestMixin):
+    SUPPORTED_ARCHITECTURES = [
+        "albert",
+        "bert",
+        "camembert",
+        # "convbert",  # accuracy off compared to pytorch: atol=1e-2
+        # "deberta",  # INF2 only
+        # "deberta_v2",  # INF2 only
+        "distilbert",
+        "electra",
+        # "flaubert",  # accuracy off compared to pytorch (not due to the padding)
+        "mobilebert",
+        "roberta",
+        "roformer",
+        # "xlm",  # accuracy off compared to pytorch (not due to the padding)
```

Comment:
Let's create an issue afterwards to keep track of the problematic models.

Reply:
Issues logged here: #90